### PR TITLE
feat: I can hide/show filter bar and add filters directly from filter bar

### LINKED ISSUE #889: I can hide/show filter bar and add filters directly from filter bar
I should be able to hide/show the filter bar by clicking either on the "Filter" button when there is an active "Filter".

When there is no "Filter" clicking on the filter button should keep the actual behavior: First, the menu opens to set the filter, then the filter bar appears.

If the filter bar is hidden and there is an active sort It should both open the filter bar and a menu next to the sort button to add a secondary sorting.

https://github.com/twentyhq/twenty/assets/19412894/6a194450-b997-4b82-9725-3f7c591b949e

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #889 

https://github.com/twentyhq/twenty/assets/140154534/a0982cfe-b816-4200-8fd9-ab273413cdf4

### CODE REVIEW

File: front/src/modules/ui/table/table-header/components/TableHeader.tsx [hunk 1]
```diff
@@ -72,12 +72,15 @@ export function TableHeader<SortField>({
           <FilterDropdownButton
             context={TableContext}
             HotkeyScope={FiltersHotkeyScope.FilterDropdownButton}
+            isPrimaryButton
```
Comment (charlesBochet) [inline front/src/modules/ui/table/table-header/components/TableHeader.tsx:75]: I'm not sure to understand what `isPrimaryButton` means?

Comment (gitstart-twenty) [reply front/src/modules/ui/table/table-header/components/TableHeader.tsx:75]: There are two filter buttons available: one located at the top and the other within the sort and filter bar. The top button possesses distinct attributes. For instance, its behavior upon clicking varies based on whether a filter already exists or not. The button in the sort-and-filter bar is a reused version of the top button. However, its behavior remains consistent, simply adding a filter without considering the presence of an existing one. To differentiate between these two buttons, we refer to the main button as `isPrimaryButton`.

File: front/src/modules/ui/filter-n-sort/components/SortDropdownButton.tsx [hunk 2]
```diff
@@ -95,6 +110,7 @@ export function SortDropdownButton<SortField>({
                 onClick={() => {
```
Comment (lucasbordeau) [inline front/src/modules/ui/filter-n-sort/components/SortDropdownButton.tsx:110]: Put this in a handleMenuItemClick and put your setIsUnfolded(true) here maybe ?

File: front/src/modules/ui/table/table-header/components/TableHeader.tsx (hunk 1 above)
Comment (lucasbordeau) [reply front/src/modules/ui/table/table-header/components/TableHeader.tsx:75]: I think it's ok for now but it's not ideal.

**Comment by lucasbordeau** [comment]
I feel like the behavior is not ideal, @Bonapara we have a UX conflict here, when I have filters my filter dropdown turns into a sort and filter bar toggle.

**Comment by lucasbordeau** [comment]
Bug when I click on filter, the sort dropdown menu opens : 

Uploading Enregistrement de l’écran 2023-08-14 à 18.59.41.mov…

**Comment by lucasbordeau** [comment]
UI : add a gap between filters and + filter 
image

**Comment by Bonapara** [comment]
> I feel like the behavior is not ideal, @Bonapara we have a UX conflict here, when I have filters my filter dropdown turns into a sort and filter bar toggle.

Can we get inspiration from Notion @lucasbordeau or revert to previous behavior?

**Comment by gitstart-twenty** [comment]
> 

@lucasbordeau This is not a bug, this was requested by the issue description

**Comment by gitstart-twenty** [comment]
> 

> > Bug when I click on filter, the sort dropdown menu opens :
> 
> @lucasbordeau This is not a bug, this was requested by the issue description. We're however open to reverting this.
> 
> > If the filter bar is hidden and there is an active sort It should both open the filter bar and a menu next to the sort button to add a secondary sorting.

File: front/src/modules/ui/filter-n-sort/components/SortDropdownButton.tsx [hunk 3]
```diff
@@ -47,6 +53,15 @@ export function SortDropdownButton<SortField>({
     setSelectedSortDirection('asc');
   }, []);
 
+  const [isSortAndFilterBarOpen, setIsSortAndFilterBarOpen] =
+    useRecoilScopedState(sortAndFilterBarState, context);
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (gitstart-twenty) [reply front/src/modules/ui/filter-n-sort/components/SortDropdownButton.tsx]: This is actually listening to changes in `isSortAndFilterBarOpen` and opening the sort menu if there's a sort selected.

> If the filter bar is hidden and there is an active sort It should both open the filter bar and a menu next to the sort button to add a secondary sorting.

**Comment by Bonapara** [comment]
> > Bug when I click on filter, the sort dropdown menu opens :
> 
> @lucasbordeau This is not a bug, this was requested by the issue description. We're however open to reverting this.
> 
> > If the filter bar is hidden and there is an active sort It should both open the filter bar and a menu next to the sort button to add a secondary sorting.
> 
> cc @Bonapara

I see, my bad @gitstart-twenty, I was not clear enough. I explained the desired behavior to @charlesBochet which is implementing it.

**Review by charlesBochet — APPROVED** [review]
@gitstart-twenty I've discussed with @Bonapara, there are a few things we want to change. Should be lite. For simplicity, I will merge as it is an open a small PR that will link here with the changes. Overall it's close to the behavior we want but the devil is in the details and it's hard to explain it here!

**Pull request merged**